libcudacxx/include/cuda/std/__exception/cuda_error.h

```diff
@@ -27,41 +27,54 @@
 #endif // !_CCCL_CUDA_COMPILER_NVCC && !_CCCL_CUDA_COMPILER_NVHPC
 
 #include <cuda/std/__exception/terminate.h>
-#include <cuda/std/detail/libcxx/include/stdexcept>
 
-#include <nv/target>
 #if !defined(_CCCL_COMPILER_NVRTC)
 #  include <cstdio>
+#  include <stdexcept>
 #endif // !_CCCL_COMPILER_NVRTC
 
+#include <nv/target>
+
 _LIBCUDACXX_BEGIN_NAMESPACE_CUDA
 
 /**
  * @brief Exception thrown when a CUDA error is encountered.
  */
-class cuda_error : public _CUDA_VSTD_NOVERSION::runtime_error
+
+#ifndef _LIBCUDACXX_NO_EXCEPTIONS
+class cuda_error : public ::std::runtime_error
 {
-public:
-  cuda_error(::cudaError_t __status, const char* __msg) noexcept
-      : _CUDA_VSTD_NOVERSION::runtime_error("")
+private:
+  static char* __format_string(::cudaError_t __status, const char* __msg, , char __formatted_msg[256]) noexcept
   {
-    ::snprintf(
-      const_cast<char*>(this->what()), _CUDA_VSTD::__libcpp_refstring::__length, "cudaError %d: %s", __status, __msg);
+    ::snprintf(__formatted_msg, 256, "cudaError %d: %s", __status, __msg);
+    return __formatted_msg;
   }
+
+public:
+  cuda_error(::cudaError_t __status, const char* __msg, char __formatted_msg[256] = {0}) noexcept
+      : ::std::runtime_error(__format_string(__status, __msg, __formatted_msg))
+  {}
```

Comment:
Critical: This is still equivalent to 
```suggestion
  cuda_error(::cudaError_t __status, const char* __msg, char* __formatted_msg = {0}) noexcept
      : ::std::runtime_error(__format_string(__status, __msg, __formatted_msg))
  {}
```
so creates only a nullptr. If you don't want to include `cuda::std::array`, you could do:
```suggestion
  struct __msg_storage {
    char __buffer[256];
  };
  
  cuda_error(::cudaError_t __status, const char* __msg, __msg_storage __formatted_msg = {}) noexcept
      : ::std::runtime_error(__format_string(__status, __msg, __formatted_msg.__buffer))
  {}
```